### ISSUE TITLE
XSS vulnerability in igv.js

### ISSUE THREAD

**Issue by srcc-chekh** [igvteam/igv.js#1687]
Some of the user text input fields do not sanitize their input.  So you can put in some HTML and cause the browser to display content from external sites.

For example for the "track name" you can put in something like:
```

"><img src=_url_ onerror=prompt()>

```

and have the browser display an external image.

I'm not actually sure how serious this problem is, but here was the video of the "exploit" I was provided:
https://drive.google.com/file/d/1Yb9vtsgd7S95b9fB94FeGE9cj9vIWaXh/view?usp=sharing

It seems like user text input fields needs some additional sanitization?

**Comment by srcc-chekh** [comment]
Claude 2 suggests:
"Test sanitization by attempting to submit malicious payloads like <script>alert(1)</script> to ensure they are encoded properly and safely rendered."

**Comment by jrobinso** [comment]
Interesting, thanks for the tip.  We will take care of it.   Although to enter text into our forms you already have physical access to the machine and can run anything your want without igv.js.

**Comment by jrobinso** [comment]
This should be easy to fix, we have only a few widgets where text is entered.

For my education, how could this be exploited without having physical access to the machine and permissions to run the web browser, and hence the ability to run anything with or without igv.js?  I'm trying to get a sense of what the actual threat is.

**Comment by srcc-chekh** [comment]
I believe igv.js is a javascript component included in many web sites?

**Comment by jrobinso** [comment]
Yes it is.  My question is how do you exploit this if you aren't physically running the browser on the client machine and entering text into the popup?

**Comment by jrobinso** [comment]
This should be fixed in version 2.15.11.   User input is run through DOMPurify.santize (https://github.com/cure53/DOMPurify) before modifying the DOM.  WRT displaying external images with a track label, this would be unusual but DOMPurify allows it.  It would be unusual but is it unsafe?   According to the DOMPurify developers it is not.  Its our intent that track labels allow for rich html, images are unusual but not necessarily disallowed.

Example transformations from https://github.com/cure53/DOMPurify

```
DOMPurify.sanitize('<img src=x onerror=alert(1)//>'); // becomes <img src="x">
DOMPurify.sanitize('<svg><g/onload=alert(2)//<p>'); // becomes <svg><g></g></svg>
DOMPurify.sanitize('<p>abc<iframe//src=jAva&Tab;script:alert(3)>def</p>'); // becomes <p>abc</p>
DOMPurify.sanitize('<math><mi//xlink:href="data:x,<script>alert(4)</script>">'); // becomes <math><mi></mi></math>
DOMPurify.sanitize('<TABLE><tr><td>HELLO</tr></TABL>'); // becomes <table><tbody><tr><td>HELLO</td></tr></tbody></table>
DOMPurify.sanitize('<UL><li><A HREF=//google.com>click</UL>'); // becomes <ul><li><a href="//google.com">click</a></li></ul>
```
